### PR TITLE
Modified `MarginalDistribution`

### PR DESCRIPTION
**Pull request by akash9712**
#### Brief description of what is fixed or changed
Modified `MarginalDistribution` so that it can be initialised without
the the use of `RandomSymbols`, using a joint distribution and indices.
#### Release Notes

NO ENTRY

### CODE REVIEW

File: sympy/core/tests/test_args.py [hunk 1]
```diff
@@ -1054,10 +1054,10 @@ def test_sympy__stats__joint_rv__JointDistributionHandmade():
 
 
 def test_sympy__stats__joint_rv__MarginalDistribution():
-    from sympy.stats.rv import RandomSymbol
+    from sympy.stats import Normal
     from sympy.stats.joint_rv import MarginalDistribution
-    r = RandomSymbol(S('r'))
-    assert _test_args(MarginalDistribution(r, (r,)))
+    N1 = Normal('N1', [1, 0], [[1, 0], [0, 1]])
+    assert _test_args(MarginalDistribution(N1, (N1[1],)))
```
Comment (Upabjojr) [inline sympy/core/tests/test_args.py]: I would prefer to construct `MarginalDistribution` without occurrences of `RandomSymbol` in it.
Like:
```python
dist = N1.pspace.distribution
_test_args(MarginalDistribution(dist, (1,)))
```
Distribution objects should not contain `RandomSymbol` or `PSpace` objects, only the parameters defining the distribution.

File: sympy/stats/joint_rv_types.py [hunk 2]
```diff
@@ -254,3 +253,50 @@ def NormalGamma(syms, mu, lamda, alpha, beta):
     A random symbol
     """
     return multivariate_rv(NormalGammaDistribution, syms, mu, lamda, alpha, beta)
+
+#-----------------------STOOCHASTIC PROCESSES--------------------------------
```
Comment (Upabjojr) [inline sympy/stats/joint_rv_types.py]: What about a new file?

File: sympy/stats/joint_rv.py [hunk 3]
```diff
@@ -330,3 +351,12 @@ def marginalise_out(self, expr, rv):
 
     def __call__(self, *args):
         return self.pdf(*args)
+
+class StochasticProcess(JointDistribution):
```
Comment (Upabjojr) [inline sympy/stats/joint_rv.py]: I don't think this is a joint distribution in the end. I should behave like a joint distribution in some cases.

File: sympy/stats/joint_rv.py [hunk 4]
```diff
@@ -330,3 +351,12 @@ def marginalise_out(self, expr, rv):
...
+    """
+    Class representing stochastic processes.
+    """
+
+    @property
+    def component_count(self):
```
Comment (Upabjojr) [inline sympy/stats/joint_rv.py]: We may have countable and uncountable stochastic processes. We need to consider that.

File: sympy/stats/joint_rv.py (hunk 3 above)
Comment (jksuom) [reply sympy/stats/joint_rv.py]: I think it should rather be like collection of joint distributions. It should have a method creating a joint distribution for each finite set of indices.

**Comment by czgdp1807** [comment]
Since, `stats` module plans to implement this concept in another way, I am closing this PR for now. However, feel free to reuse it in your GSoC project. Thanks for the contributions.

**Pull request closed without merging**